lib/jsdom/browser/Window.js

```diff
@@ -183,23 +186,42 @@ function Window(options) {
   ///// METHODS
 
   let latestTimerId = 0;
+  let latestAnimationFrameCallbackId = 0;
 
   this.setTimeout = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, window.__timers, args);
   };
   this.setInterval = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, window.__timers, args);
+  };
+  this.requestAnimationFrame = fn => {
+    const hr = process.hrtime(windowInitialized);
+    const hrInNano = hr[0] * 1e9 + hr[1];
+    const fps = 1000 / 60;
+    const hrInMicro = hrInNano / 1e6;
+
+    return startTimer(
+      window,
+      setTimeout,
+      clearTimeout,
+      ++latestAnimationFrameCallbackId,
+      fn,
+      fps,
+      window.__animationFrameCallbacks,
+      [hrInMicro]
+    );
   };
-  this.clearInterval = stopTimer.bind(this, window);
-  this.clearTimeout = stopTimer.bind(this, window);
+  this.clearInterval = stopTimer.bind(this, window.__timers);
+  this.clearTimeout = stopTimer.bind(this, window.__timers);
+  this.cancelAnimationFrame = stopTimer.bind(this, window.__animationFrameCallbacks);
   this.__stopAllTimers = stopAllTimers.bind(this, window);
```

Comment:
You need to cancel all animation frame callbacks `window.close()` as well.

Reply:
Ah, I was unsure of the purpose here. Will do, thanks!